### PR TITLE
Authentication from yaml implementation #146

### PR DESCRIPTION
**Pull request by olenagerasimova**
Started authentication from yaml implementation for #146 by creating test. Also upgraded http module dependency and added todo to continue.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#157](https://github.com/artipie/artipie/issues/157) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/c04847d5-fdf5-49fa-ba17-9b12a88e0ba0); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@g4s8 please merge

File: src/main/java/com/artipie/YamlAuth.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.artipie.http.auth.Authentication;
+import java.util.Optional;
+import org.apache.commons.lang3.NotImplementedException;
+
+/**
+ * Authentication implementation based on yaml file with credentials.
+ * @since 0.3
+ * @todo #146:30min Implements this class to find user by credentials and enable test. For more
+ *  details check #146 and note that password is Base64 encoded.
```
Comment (g4s8) [inline src/main/java/com/artipie/YamlAuth.java:35]: @olenagerasimova why base64? Standard tools such as `sha256sum` uses plain hex encoding for sha256

Comment (olenagerasimova) [reply src/main/java/com/artipie/YamlAuth.java:35]: @g4s8 according to java doc of `com.artipie.http.auth.Authentication#user` param `password` is Base64 encoded. Or I've got something wrong? If it's not encoded, I should correct the test

Comment (g4s8) [reply src/main/java/com/artipie/YamlAuth.java:35]: @olenagerasimova the javadoc of `com.artipie.http.auth.Authentication#user` is wrong - I just tested that it's working correctly with plain text passwords (when using http-basic)

**Comment by olenagerasimova** [comment]
@g4s8 thanks, I've corrected todo and test. I'll create PR to correct javadoc in 10mins to avoid future misunderstanding.

**Comment by 0crat** [comment]
Job was finished in 20 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Pull request merged**